sdk/core/Azure.Core/src/Pipeline/Policies/LoggingPolicy.cs

```diff
@@ -133,11 +134,25 @@ private static async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemo
             {
                 if (textResponse)
                 {
-                    await s_eventSource.ResponseContentTextAsync(message.Response, responseTextEncoding, message.CancellationToken).ConfigureAwait(false);
+                    if (async)
+                    {
+                        await s_eventSource.ResponseContentTextAsync(message.Response, responseTextEncoding, message.CancellationToken).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        s_eventSource.ResponseContentText(message.Response, responseTextEncoding, message.CancellationToken);
```

Comment:
Why do we need these sync code paths? We are inside an async method.

Reply:
It's a pattern we use where we write a method that takes a `bool async` parameter and becomes fully sync or async depending on the value.

This is a shorter example from BearerTokenAuthenticationPolicy implementation:


``` C#
        public override Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
        {
            return ProcessAsync(message, pipeline, true);
        }

        public override void Process(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
        {
            ProcessAsync(message, pipeline, false).EnsureCompleted();
        }

        public async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline, bool async)
        {
>>>>        string token = async ?
                    await _credential.GetTokenAsync(_scopes, message.CancellationToken).ConfigureAwait(false) :
                    _credential.GetToken(_scopes, message.CancellationToken);

            if (token != _currentToken)
            {
                // Avoid per request allocations
                _currentToken = token;
                _headerValue = "Bearer " + token;
            }

            message.Request.SetHeader(HttpHeader.Names.Authorization, _headerValue);

            if (async)
            {
>>>>            await ProcessNextAsync(message, pipeline);
            }
            else
            {
>>>>            ProcessNext(message, pipeline);
            }
        }
```